From: Iain Lane <laney@debian.org>
Date: Mon, 18 Nov 2019 10:42:39 +0000
Subject: Revert "glocalfileinfo: Only return file mode, not type,
 as UNIX_MODE attribute"

This reverts commit bfdc5fc4fc84ef8518d2d1a328c8482cf5a38e98.

This commit changes the semantics of the `unix::mode` attribute, which
some things (the one we've noticed is ostree) rely on.

Bug-Upstream: https://gitlab.gnome.org/GNOME/glib/issues/1934
---
 gio/glocalfileinfo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/glocalfileinfo.c b/gio/glocalfileinfo.c
index 5ba7691..5245567 100644
--- a/gio/glocalfileinfo.c
+++ b/gio/glocalfileinfo.c
@@ -987,7 +987,7 @@ set_info_from_stat (GFileInfo             *info,
   /* Mostly pointless on Windows.
    * Still, it allows for S_ISREG/S_ISDIR and IWRITE (read-only) checks.
    */
-  _g_file_info_set_attribute_uint32_by_id (info, G_FILE_ATTRIBUTE_ID_UNIX_MODE, statbuf->st_mode & ~S_IFMT);
+  _g_file_info_set_attribute_uint32_by_id (info, G_FILE_ATTRIBUTE_ID_UNIX_MODE, statbuf->st_mode);
 #if defined (HAVE_STRUCT_STAT_ST_BLKSIZE)
   _g_file_info_set_attribute_uint32_by_id (info, G_FILE_ATTRIBUTE_ID_UNIX_BLOCK_SIZE, statbuf->st_blksize);
 #endif
